tests/functional/example/test_scikit_learn.py

```diff
@@ -0,0 +1,65 @@
+"""Tests the minimalist example script on scitkit-learn and its integration to Oríon."""
+import os
+import subprocess
+
+import pytest
+
+from orion.client import create_experiment
+import orion.core.cli
+from orion.storage.base import get_storage
+
+
+def test_script_integrity(capsys):
+    """Verifies the example script can run in standalone via `python ...`."""
+    script = os.path.abspath("examples/scikitlearn-iris/main.py")
+
+    return_code = subprocess.call(["python", script, '0.1'])
+
+    assert return_code != 2, "The example script does not exists."
+    assert return_code != 1, "The example script did not terminates its execution."
+    assert return_code == 0 and not capsys.readouterr().err, \
+        "The example script encountered an error during its execution."
+
+
+@pytest.mark.usefixtures("clean_db")
+@pytest.mark.usefixtures("null_db_instances")
+def test_orion_runs_script(monkeypatch, database):
+    """Verifies Oríon can execute the example script."""
+    script = os.path.abspath("examples/scikitlearn-iris/main.py")
+    monkeypatch.chdir(os.path.dirname(os.path.abspath(__file__)))
+    config = "orion_config.yaml"
+
+    orion.core.cli.main(["hunt", "--config", config, "python", script,
+                         "orion~choices([0.1])"])
+
+    exps = list(database.experiments.find({'name': 'scikit-iris-tutorial'}))
+    assert len(exps) == 1
+
+    exp = exps[0]
+    assert exp['version'] == 1
+    assert len(exp['space']) == 1
+    assert '/_pos_2' in exp['space']
+
+    storage = get_storage()
+    trials = storage.fetch_trials(uid=exp['_id'])
+    assert len(trials) == 1
+
+    trial = trials[0]
+    assert trial.status == 'completed'
+    assert trial.params['/_pos_2'] == 0.1
+
+
+@pytest.mark.usefixtures("clean_db")
+@pytest.mark.usefixtures("null_db_instances")
+def test_script_results(monkeypatch):
```

Comment:
Do we need this test are the one above is sufficient? I don't see the difference beside one looking at experiment.stats, but that could be done in the same test. :thinking:

Reply:
I'm in favor of keeping them separate because they tests different things: the second one tests that the script can run to completion with Oríon. The third one tests that the script gives replicable results (which is important in the context of a tutorial). That way, if there is a replication issue, only the third test will fail instead of the one signaling that the script is incompatible with Oríon. 

However, we may improve the name of the test to reflect its purpose in case the docstring is not enough.

Reply:
Sounds reasonable. I suggest you rename the last test `test_result_reproducibility`.

Reply:
Perfect!